tarsier/tag_utils.ts

```diff
@@ -162,36 +162,69 @@ window.tagifyWebpage = (tagLeafTexts = false) => {
       continue;
     }
 
```

Comment:
Functions a bit long and hard to read. Ideally we broke this apart but given this is a js utility file in a python project this should be fine.